LEAF_Request_Portal/templates/print_subindicators_ajax.tpl

```diff
@@ -1,12 +1,19 @@
 <!--{**}-->
         <!--{if $indicator.is_sensitive == 1}-->
-                <div class="sensitiveIndicatorMaskToggle">
-                    <input type="checkbox" id="sensitiveIndicatorMaskCheckbox_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}-->" onClick="toggleSensitiveIndicator(<!--{$indicator.indicatorID|strip_tags}-->, <!--{$indicator.series|strip_tags}-->, this.checked);">
-                    <label for="sensitiveIndicatorMaskCheckbox_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}-->" title="Show Sensitive Data" alt="Show Sensitive Data"  tabindex="0" onkeydown="if (event.keyCode==13){ this.click(); }"></label>
-                </div>
-                <span class="sensitiveIndicator-masked">
-                    *****
-                </span>
+            <div class="sensitiveIndicatorMaskToggle">
+                <input type="checkbox" id="sensitiveIndicatorMaskCheckbox_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}-->" onClick="toggleSensitiveIndicator(<!--{$indicator.indicatorID|strip_tags}-->, <!--{$indicator.series|strip_tags}-->, this.checked);">
+                <label for="sensitiveIndicatorMaskCheckbox_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}-->" title="Show Sensitive Data" alt="Show Sensitive Data"  tabindex="0" onkeydown="if (event.keyCode==13){ this.click(); }"></label>
+            </div>
+            <span class="sensitiveIndicator-masked">
+                *****
+            </span>
+            <script>
+                $("div.sensitiveIndicator#xhrIndicator_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}-->").hover(function() {
+                    $("div.sensitiveIndicator#xhrIndicator_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}--> > div.sensitiveIndicatorMaskToggle > label").trigger('click');
+                }, function() {
+                    $("div.sensitiveIndicator#xhrIndicator_<!--{$indicator.indicatorID|strip_tags}-->_<!--{$indicator.series|strip_tags}--> > div.sensitiveIndicatorMaskToggle > label").trigger('click');
+                })
+            </script>
```

Comment:
A few things:
1.) If an indicator has any children, the hidden text is toggled when you hover over any of the child elements, this is especially noticeable when you hover over a sensitive child indicator to a sensitive parent (they both reveal at the same time). I would just make the selector more specific.
2.) On the printview page - if you edit a field, the toggle stops working. This is because the div is removed and re-added wit javascript, so the event is not bound to the new element. To bind to dynamically added elements, you should use the .on function in jquery. This link will show you how, specifically the edit from DECEMBER 11, 2012 (https://stackoverflow.com/questions/9827095/is-it-possible-to-use-jquery-on-and-hover).
3.) If you are hovering, the field becomes visible. Then you click the eye, it becomes invisible. Then you un-hover and it becomes permanently visible. I'm not sure how we want the click to effect the visibility in the end, but I would suggest adding a param to the toggle function that takes a visibility state, on or off basically. And if you mouseover toggle(thisid, on), if you mouseout toggle(thisid, off).

Reply:
Addressed 1 and 2 in commit `a89`.  @mgaoVA please give clarification on what should be done with 3.

Reply:
For 3, If the indicator has been clicked to be shown, the hover effect should go away. In other words, if it is shown, hovering over it does nothing. The hover effect should only happen if the indicator is hidden.

Reply:
Addressed in commit `b5f`